CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
+project(zstr)
+
+# zlib is required
+FIND_PACKAGE(ZLIB 1.2.3 REQUIRED)
+include_directories(BEFORE ${ZLIB_INCLUDE_DIRECTORIES})
+GET_FILENAME_COMPONENT(ZLIB_BASEPATH ${ZLIB_INCLUDE_DIRS} PATH)
+LINK_DIRECTORIES(${ZLIB_BASEPATH}/lib64 ${ZLIB_BASEPATH}/lib)
+MESSAGE("Adding ZLIB for ${PROJECT_NAME}: -I${ZLIB_INCLUDE_DIRS} and -L${ZLIB_BASEPATH}/lib64 -L${ZLIB_BASEPATH}/lib)")
```

Comment:
Could you use instead target_include and target_link as suggested here?
http://mariobadr.com/creating-a-header-only-library-with-cmake.html
https://stackoverflow.com/questions/60604249/how-to-make-a-header-only-library-with-cmake
More complex options e.g. at: https://github.com/nlohmann/json/blob/develop/CMakeLists.txt

Reply:
Sure.  I wanted it to be compatible with cmake 3.10.2 (i.e. ubuntu 18) for which interface libraries are broken, but it is easy enough to add a warning and the work-around necessary for versions below 3.13.